diff --git a/src/llvm/projects/libtapi/include/tapi/LinkerInterfaceFile.h b/src/llvm/projects/libtapi/include/tapi/LinkerInterfaceFile.h
--- a/src/llvm/projects/libtapi/include/tapi/LinkerInterfaceFile.h
+++ b/src/llvm/projects/libtapi/include/tapi/LinkerInterfaceFile.h
@@ -14,6 +14,7 @@
 #define TAPI_LINKER_INTERFACE_FILE_H
 
 #include <memory>
+#include <stdint.h>
 #include <string>
 #include <tapi/Defines.h>
 #include <vector>
diff --git a/src/llvm/projects/libtapi/include/tapi/PackedVersion32.h b/src/llvm/projects/libtapi/include/tapi/PackedVersion32.h
--- a/src/llvm/projects/libtapi/include/tapi/PackedVersion32.h
+++ b/src/llvm/projects/libtapi/include/tapi/PackedVersion32.h
@@ -13,6 +13,7 @@
 #ifndef TAPI_PACKED_VERSION_32_H
 #define TAPI_PACKED_VERSION_32_H
 
+#include <stdint.h>
 #include <tapi/Defines.h>
 
 ///
